src/goto-programs/builtin_functions.cpp

```diff
@@ -699,7 +700,7 @@ void goto_convertt::do_havoc_slice(
   t->source_location_nonconst().set_comment(
     "assertion havoc_slice " + from_expr(ns, identifier, ok_expr));
 
-  const array_typet array_type(char_type(), arguments[1]);
+  const array_typet array_type(char_type(), simplify_expr(arguments[1], ns));
```

Comment:
I get that this might be the right thing to do and that it fixes a bug but this doesn't immediately seem linked to the description of the PR.

Reply:
Thanks, addressed in two ways: 1) I've moved this change to a commit of its own; 2) I've added a comment explaining when a `DATA_INVARIANT` newly introduced here could fail (which really was what made me detect the above).

Reply:
Amazing!  Thank you.